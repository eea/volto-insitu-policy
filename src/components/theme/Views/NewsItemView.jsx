import React from 'react';
import RenderBlocks from '@plone/volto/components/theme/View/RenderBlocks';
import { HTMLField } from '@eeacms/volto-insitu-policy/helpers';
import './styles.less';

function NewsItemView(props) {
  const { content } = props;

  return (
    <div className="insitu-newsitem-view">
      <div className="ui container">
        <p className="documentDescription eea callout">{content.description}</p>
        {!!content.image && (
          <div className="lead-img">
            <img
              className="main-img"
              src={content.image?.download}
              alt={content.title}
            />
            <p className="main-img-caption">{content.image_caption}</p>
          </div>
        )}
        <HTMLField value={content.text} />
        {content.blocks_layout.items && <RenderBlocks {...props} />}
      </div>
    </div>
  );
}

export default NewsItemView;
