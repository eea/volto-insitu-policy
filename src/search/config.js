import { mergeConfig } from '@eeacms/search';
import { build_runtime_mappings } from '@eeacms/volto-globalsearch/utils';
import { getClientProxyAddress, getSearchThumbUrl } from './utils';
import vocabs from './vocabulary';
import facets from './facets';

const insituConfig = {
  title: 'Insitu Main',
};

export const clusters = {
  name: 'op_cluster',
  field: 'objectProvides',
  clusters: [
    {
      name: 'News',
      values: ['News'],
      defaultResultView: 'horizontalCard',
    },
    {
      name: 'Use Cases',
      values: ['insitu.use_case'],
      defaultResultView: 'horizontalCard',
    },
    {
      name: 'Reports',
      values: ['insitu.report'],
      defaultResultView: 'horizontalCard',
    },
    {
      name: 'Others',
      values: ['Webpage', 'Dashboard', 'Event', 'Link'],
      defaultResultView: 'horizontalCard',
    },
  ],
};

export default function install(config) {
  const envConfig = process.env.RAZZLE_ENV_CONFIG
    ? JSON.parse(process.env.RAZZLE_ENV_CONFIG)
    : insituConfig;

  const pjson = require('@eeacms/volto-insitu-policy/../package.json');

  envConfig.app_name = pjson.name;
  envConfig.app_version = pjson.version;

  config.searchui.insituSearch = {
    ...mergeConfig(envConfig, config.searchui.globalsearchbase),
    elastic_index: '_es/insituSearch',
    index_name: 'copernicus_searchui',
    host: process.env.RAZZLE_ES_PROXY_ADDR || 'http://localhost:3000',
    runtime_mappings: build_runtime_mappings(clusters),
    ...vocabs,
  };

  const { insituSearch } = config.searchui;

  insituSearch.permanentFilters.push({
    term: {
      cluster_name: 'copernicus_insitu',
    },
  });

  insituSearch.initialView.tilesLandingPageParams.sections = [
    {
      id: 'types',
      title: 'Types',
      facetField: 'objectProvides',
      sortOn: 'alpha',
      icon: {
        family: 'Content types',
      },
    },
  ];

  insituSearch.facets = facets;

  insituSearch.contentSectionsParams = {
    enable: true,
    sectionFacetsField: 'op_cluster',
    sections: clusters.clusters,
    clusterMapping: Object.assign(
      {},
      ...clusters.clusters.map(({ name, values }) =>
        Object.assign({}, ...values.map((v) => ({ [v]: name }))),
      ),
    ),
  };

  insituSearch.resultItemModel = {
    factory: 'ResultModel',
    urlField: 'about',
    titleField: 'title',
    metatypeField: 'objectProvides',
    descriptionField: 'description',
    tagsField: 'topic',
    issuedField: 'issued',
    getThumbnailUrl: 'getSearchThumbUrl',
    getIconUrl: 'getGlobalsearchIconUrl',
    fallbackThumbUrl:
      'https://react.semantic-ui.com/images/wireframe/white-image.png',
  };

  config.resolve.getSearchThumbUrl = getSearchThumbUrl();

  if (typeof window !== 'undefined') {
    config.searchui.insituSearch.host =
      process.env.RAZZLE_ES_PROXY_ADDR || getClientProxyAddress();
  }

  return config;
}
