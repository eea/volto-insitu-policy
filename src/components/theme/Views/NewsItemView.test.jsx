import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import NewsItemView from './NewsItemView';
import { render } from '@testing-library/react';
import config from '@plone/volto/registry';

config.blocks = {
  blocksConfig: {
    title: {
      view: () => <div>Title Block Component</div>,
    },
  },
};

const mockStore = configureStore();

jest.mock('semantic-ui-react', () => ({
  ...jest.requireActual('semantic-ui-react'),
}));

describe('NewsItemView', () => {
  it('should render the component', () => {
    const content = {
      title: 'NewsItemView',
    };

    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const { getByText } = render(
      <Provider store={store}>
        <MemoryRouter>
          <NewsItemView content={content} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByText('Title Block Component')).toBeInTheDocument();
    // expect(getByText('Test test')).toHaveClass('test-class');
  });
});
