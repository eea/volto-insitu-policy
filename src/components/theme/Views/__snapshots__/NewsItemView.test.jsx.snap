// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptationOptionView should render the component 1`] = `
<div
  className="insitu-newsitem-view"
>
</div>
`;
